server/util/lockingbuffer/lockingbuffer.go

```diff
@@ -34,3 +35,11 @@ func (lb *LockingBuffer) Read(p []byte) (int, error) {
 	defer lb.mu.Unlock()
 	return lb.buffer.Read(p)
 }
+
+// ReadAll provides a thread-safe alternative to io.ReadAll for lockingbuffer.
+func (lb *LockingBuffer) ReadAll() ([]byte, error) {
+	lb.mu.Lock()
+	defer lb.mu.Unlock()
+	defer lb.buffer.Reset()
+	return io.ReadAll(bytes.NewReader(lb.buffer.Bytes()))
```

Comment:
maybe just un-defer this so it's clearer to readers?

buf, err := io.ReadAll(bytes.NewReader(lb.buffer.Bytes()))
lb.buffer.Reset()
return buf, err

Reply:
sg, yeah with the defer it takes a bit longer to grok that the Reset happens with the lock still kept. Done